### PR TITLE
Makes felinids die from chocolate.

### PR DESCRIPTION
**Pull request by GuyonBroadway**
This pr is held up by: https://github.com/tgstation/tgstation/issues/46236

I know felinids are a deprecated feature and we probably should not be supporting them but..

:cl: GuyonBroadway
balance: Felinids now have a very adverse reaction to chocolate products.
/:cl:

image

The people have spoken. 

Felinids now vomit, become nauseated and suffer major organ damage from consuming chocolate.

### CODE REVIEW

**Comment by Naloacisamazing** [comment]
Fucking wew lad

**Comment by ExcessiveUseOfCobblestone** [comment]
haha jk don't merge this pr it was just a joke... unless...

**Comment by Akrilla** [comment]
😳

**Comment by GuyonBroadway** [comment]
Sadly this "do not merge" is due to a bug that means if someone vomits all the food reagents in their system vanish.

This means all that happens is a cat pukes and is fine because all the chocolate leaves their system.

**Comment by Takov-Mala** [comment]
> 
> 
> Sadly this "do not merge" is due to a bug that means if someone vomits all the food reagents in their system vanish.
> 
> This means all that happens is a cat pukes and is fine because all the chocolate leaves their system.

You should still be able to give them some toxin damage as vomiting usually iirc does.

**Comment by GuyonBroadway** [comment]
Raw toxin damage is boring, total organ failure is much more interesting.

**Comment by GuyonBroadway** [comment]
Closing pending maintainer approval.

**Comment by GuyonBroadway** [comment]
image

**Comment by Sishen1542** [comment]
porting this to citadel asap

**Comment by GuyonBroadway** [comment]
You should stop porting my PR's to citadel, they are NOT good ideas.

**Comment by Sishen1542** [comment]
> You should stop porting my PR's to citadel, they are NOT good ideas.

you're right, they're great ideas :sunglasses:

**Comment by ExcessiveUseOfCobblestone** [comment]
Any changes to species specific interactions, including availability of said species, should be okayed by a maintainer prior to PRing imo (that's how I been enforcing it at least).

This is funny enough I'll okay it just to see if someone merges.

**Comment by GuyonBroadway** [comment]
Really I need to find out why vomiting purges any and all food reagents beforehand. But I couldn't resist.


Oh and for the record while this IS a webedit PR I did compile and run it locally.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (nemvar) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Taser cocolate :eyes:

Comment (PKPenguin321) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: yee

Comment (GuyonBroadway) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: You say that like it's a BAD thing.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: brilliant

**Comment by ATH1909** [comment]
> > Sadly this "do not merge" is due to a bug that means if someone vomits all the food reagents in their system vanish.
> > This means all that happens is a cat pukes and is fine because all the chocolate leaves their system.
> 
> You should still be able to give them some toxin damage as vomiting usually iirc does.

Uh, doesn't vomiting HEAL toxin damage, not DEAL it?

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (PKPenguin321) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Im dyi-wying :3

**Comment by wesoda25** [comment]
My catgirl murderbones just got a lot easier

**Comment by bgobandit** [comment]
onions/garlic are also toxic to cats, fortunately unlike actual food our codebase barely uses them

**Comment by OverlyGeneric** [comment]
It's relative to the type of chocolate and the dosage.

If we're talking about dark chocolate then it will absolutely destroy the cat from the inside but even baking chocolate is considered highly toxic to them.

I'll also list a bunch of symptoms from toxin poisoning that would be feasible for the game: lethargy, seizures, tremors, vomiting, nausea, excessive panting, increased blood pressure, coma, and death.

**Comment by necromanceranne** [comment]
I'd maybe put the chainstun behind some kind of threshold (can you even have species specific overdoses?), but otherwise this is really funny.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (bgobandit) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: You feel like you could use some helpies.

**Comment by Shadowflame909** [comment]
Chefs are finna be perma'd for murder for making chocolate cakes

**Comment by InsaneKatrina** [comment]
I'm gonna miss enjoying chocolates I guess

**Comment by lolman360** [comment]
fuck cats lmoa

everyone's forgetting this PR nerfs kevinz

**Comment by kriskog** [comment]
> 
> 
> fuck cats lmoa

UwU :3

**Comment by GuyonBroadway** [comment]
> 
> 
> I'd maybe put the chainstun behind some kind of threshold (can you even have species specific overdoses?), but otherwise this is really funny.

It only causes vomiting on first application (and a high enough disgust rating) so it shouldn't chainstun unless a very clever chemist timed their doses right.

**Comment by GuyonBroadway** [comment]
Removing DNM due to locating the cause of the bug.

**Comment by diegoxt45** [comment]
finally, a good fucking  PR

**Comment by MCterra10** [comment]
Yes?
Yes.

**Comment by MCterra10** [comment]
content

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (cebutris) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Why is this not `"You feel nyauseous"`

**Comment by FancyCyclops** [comment]
Looks like the gene mods caught up with them.

**Comment by KetasDotepasto** [comment]
Is there a way to tie this to the liver instead?

Like, if you have no liver, or a felinid liver, chocolate kills you.

**Comment by MadoFrog** [comment]
I'm glad this picture I got from giving $20 to a guy who draws vore in MS paint has brought so joy to this world.

**Comment by optimumtact** [comment]
thanosgang

**Comment by GuyonBroadway** [comment]
I'd hold off on making any desicions about this pr until this one gets resolved: https://github.com/tgstation/tgstation/pull/46241.

**Comment by kyrahabattoir** [comment]
Should also make it so sleep toxins don't work on them.

**Comment by Pacmandevil** [comment]
@kyrahabattoir is there actually something that prevents from getting knocked the fuck out by sedatives? That seems really weird but I also sniff farts for a living so who am i to talk

**Comment by optimumtact** [comment]
pikoh

**Comment by nero1024** [comment]
Make it so Ian dies from chocolate too. I've been feeding him a whole vending machine's worth of chocolate yet he's still fine.

**Comment by kyrahabattoir** [comment]
@Pacmandevil Cats don't respond to a lot of human sedatives, a few actually makes them agitated instead.

**Comment by GuyonBroadway** [comment]
You can get away with nerfing a nonhuman in the name of menes, there is no way the maintainers will let you buff a nonhuman.

Especially as felinids have the fewest downsides of any roundstart playable nonhuman species.

**Comment by kyrahabattoir** [comment]
I'm sure maintainers will do what they feel like doing. I'm just pitching in.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (AnturK) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Should be in species handle_chemicals.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (AnturK) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: This could probably use species specific proc too but i guess that's out of scope.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Review by AnturK — APPROVED** [review]

**Comment by AnturK** [comment]
:cake: :cat2:

**Comment by GuyonBroadway** [comment]
YOU FOOL! I HAND'T FINISHED TESTING IT! IT'S EVER SO SLIGHTLY BUGGED!!!

File: code/modules/mob/living/carbon/human/species_types/felinid.dm
Comment (GuyonBroadway) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm]: THIS NEEDED TO BE A TAB BACKWARD! OH LORDE IS ME!!

**Comment by MadoFrog** [comment]
🤮

**Comment by GuyonBroadway** [comment]
THIS MEANS THAT FELINIDS CAN NEVER NATURALLY PURGE CHOCOLATE FROM THEIR SYSTEMS!!

I mean its funny as FUCK but....

File: code/modules/mob/living/carbon/human/species_types/felinid.dm
Comment (AnturK) [reply code/modules/mob/living/carbon/human/species_types/felinid.dm]: Just means it will process as usual too.

**Comment by AnturK** [comment]
Open a fix please.

**Comment by GuyonBroadway** [comment]
Working on it.... this is less "working" that I thought

**Comment by FloranOtten** [comment]
Leave it. Too fun.

**Pull request merged**
